article/app/controllers/ArticleController.scala

```diff
@@ -192,18 +172,66 @@ object ArticleController extends Controller with RendersItemResponse with Loggin
   /**
    * convert a response into something we can render, and return it
    * optionally, throw a response if we know it's not right to send the content
-   * @param response
+    *
+    * @param response
    * @return Either[PageWithStoryPackage, Result]
    */
-  def redirect(response: ItemResponse)(implicit request: RequestHeader): Either[PageWithStoryPackage, Result] = {
+  def responseToModelOrResult(pageParam: Option[String])(response: ItemResponse)(implicit request: RequestHeader): Either[PageWithStoryPackage, Result] = {
     val supportedContent = response.content.filter(isSupported).map(Content(_))
-    val content: Option[PageWithStoryPackage] = supportedContent.map {
-      case minute: Article if minute.isUSMinute => MinutePage(minute, RelatedContent(minute, response))
-      case liveBlog: Article if liveBlog.isLiveBlog => LiveBlogPage(liveBlog, RelatedContent(liveBlog, response))
-      case article: Article => ArticlePage(article, RelatedContent(article, response))
+    val page = pageParam.map(ParseBlockId.apply)
+    val supportedContentResult = ModelOrResult(supportedContent, response)
+    val content: Either[PageWithStoryPackage, Result] = supportedContentResult.left.flatMap { content =>
+      (content, page) match {
+        case (minute: Article, None) if minute.isUSMinute =>
+          Left(MinutePage(minute, RelatedContent(minute, response)))
+        case (liveBlog: Article, None) if liveBlog.isLiveBlog =>
+          createLiveBlogModel(liveBlog, response, None)
+        case (liveBlog: Article, Some(Some(requiredBlockId))) if liveBlog.isLiveBlog =>
```

Comment:
the outer one is whether the user specified a block id, and the inner one was whether the one they specified was actually valid.  If they specified one which is invalid, it's a 404.  Maybe I could do some matching onto case classes or something clever, or at least add a comment though.